vars/ciMetrics.groovy

```diff
@@ -0,0 +1,51 @@
+import org.centos.pipeline.CIMetrics
+
+/*
+A class to store build metrics over the lifetime of the build.
+Metrics are stored in customDataMap and then sent to influx at
+the end of the job. Example usage:
+
+try {
+    def stepName = "mystep"
+    ciMetrics.timed stepName, {
+        stage(stepName) {
+            echo "in mystep"
+        }
+    }
+    currentBuild.result = "SUCCESS"
+} catch(err) {
+    currentBuild.result = "FAILED"
+    throw err
+} finally {
+    ciMetrics.writeToInflux()
+}
+ */
+class ciMetrics {
+
+    // A map to store the data sent to influx
+    def customDataMap = ["ci_pipeline": [:]]
+    // This will prefix the data sent to influx. Usually set to the job name.
+    def prefix = "ci_pipeline"
+    // The influx target configured in jenkins
+    def influxTarget = "localInflux"
+    // The influx database that will be written to
+    def measurement = "ci_pipeline"
```

Comment:
Could this be configured such that it is based on the job name as well? Or would it make more sense to leave it as a non-variable database name? I'm not sure what your influx database provisioning code looks like.